pkg/asset/agent/manifests/agentpullsecret.go

```diff
@@ -143,43 +152,48 @@ func normalizeDockerConfig(stringData string) (string, error) {
 
 func (a *AgentPullSecret) validatePullSecret() field.ErrorList {
 	if err := a.validateSecretIsNotEmpty(); err != nil {
-		return err
+		return field.ErrorList{err}
 	}
 
-	fieldPath := field.NewPath("StringData")
-	dockerConfig := a.Config.StringData[pullSecretKey]
+	fieldPath := field.NewPath("Data")
+	dockerConfig, ok := a.Config.StringData[pullSecretKey]
+	if !ok {
+		dockerConfig = string(a.Config.Data[pullSecretKey])
+	}
 	if err := validate.ImagePullSecret(dockerConfig); err != nil {
 		return field.ErrorList{field.Invalid(fieldPath, dockerConfig, err.Error())}
 	}
 
 	return field.ErrorList{}
 }
 
-func (a *AgentPullSecret) validateSecretIsNotEmpty() field.ErrorList {
-	var allErrs field.ErrorList
-
-	fieldPath := field.NewPath("StringData")
+func (a *AgentPullSecret) validateSecretIsNotEmpty() *field.Error {
+	fieldPath := field.NewPath("Data")
 
-	if len(a.Config.StringData) == 0 {
-		allErrs = append(allErrs, field.Required(fieldPath, "the pull secret is empty"))
-		return allErrs
+	if len(a.Config.StringData) == 0 && len(a.Config.Data) == 0 {
+		return field.Required(fieldPath, "the pull secret is empty")
 	}
 
 	pullSecret, ok := a.Config.StringData[pullSecretKey]
 	if !ok {
-		allErrs = append(allErrs, field.Required(fieldPath, "the pull secret key '.dockerconfigjson' is not defined"))
-		return allErrs
+		pullSecretBytes, ok := a.Config.Data[pullSecret]
+		if !ok {
+			return field.Required(fieldPath, "the pull secret key '.dockerconfigjson' is not defined")
+		}
+		pullSecret = string(pullSecretBytes)
 	}
 
 	if pullSecret == "" {
-		allErrs = append(allErrs, field.Required(fieldPath, "the pull secret does not contain any data"))
-		return allErrs
+		return field.Required(fieldPath, "the pull secret does not contain any data")
 	}
 
-	return allErrs
+	return nil
 }
 
 // GetPullSecretData returns the content of the pull secret.
 func (a *AgentPullSecret) GetPullSecretData() string {
-	return a.Config.StringData[".dockerconfigjson"]
+	if _, ok := a.Config.StringData[pullSecretKey]; ok {
+		return a.Config.StringData[pullSecretKey]
+	}
+	return string(a.Config.Data[pullSecretKey])
```

Comment:
We need to decode from base64 at this point.

Reply:
Surprisingly, string function converts it from base64 to string. Tried it with the integration tests and it seemed to convert. Will look into it further to confirm it.